pkg/kv/kvserver/replica_application_state_machine.go

```diff
@@ -326,7 +313,7 @@ func (sm *replicaStateMachine) handleNonTrivialReplicatedEvalResult(
 
 	// TODO(#93248): the strongly coupled truncation code will be removed once the
 	// loosely coupled truncations are the default.
-	if truncState != nil {
+	if truncState := rResult.GetRaftTruncatedState(); truncState != nil {
```

Comment:
so if we discarded the truncation from rResult earlier (in `stageSomething`), it won't show up here, right?

Reply:
Yes. The lifetime of `rResult.GetRaftTruncatedState()` ends wherever `DiscardRaftTruncation()` is called.

- For loosely coupled truncs, it ends in `stageTruncation` (because the thing is handed over to the loosely coupled machinery right away).
- For tightly coupled truncs, it ends either in `stageTruncation` (if we see that this truncation is a no-op), or here after having applied it.

I can put a comment to this extent.